course/assignments/R-intro-lectures-assignments.md

```diff
@@ -0,0 +1,60 @@
+3. What is Rmarkdown and why are we using it in this class? How do you execute a chunk of code in an Rmarkdown document?
+4. Is there anything in particular you are excited about learning about theoretical ecological population models and about using R for data analyses and modelling?
+
+# Week 2 (8 marks) Rmarkdown, assignment, functions, vectors, data frames
+All the following should be done with Rmarkdown.
+
+1. Assignment
+    a. Assign the value `5` to the variable/object `a`.
+    b. Assign the result of `10/3` to the variable `b`.
+    c. Assign the product of `a` and `b` to `c`.
+    d. Write a function that multiplies two numbers and returns the result. Use it to assign the product of `a` and `b` to `c`.
+2. Vectors
+    a. Create a vector `v` with all integers 1-30.
+    b. Use `v` to create a new vector `v_square` with the square of element 3, 6, 7, 10, 15, 22, 23, 24, and 30 from `v`. Do _not_ use a for loop.
+    c. Extract every 5th element from `v`, into a new vector `v_every_5th`.
+4. Boolean indexing
+    a. Create a boolean vector `v_bool` of the same length as `v`, indicating which vector elements are bigger than 20.
+    b. Use `v_bool` as an index to extract the elements from `v` that are bigger than 20 and save as `v_over_20`.
+    c. What's the mean of `v_over_20`?
+5. Data frames
+    a. What are the column names of inbuilt dataframe `iris`?
+    b. How can you view the first 5 rows of this data frame?
+    c. How many observations are there?
+    d. Which factor levels are there? What is a factor level and when can it be useful?
+    e. What is the mean value for the sepal.height column?
+6. Save the data frame with only the species and sepal.width columns as csv-file. Load this csv file as a new data frame and run display its summary statistics.
+
+
+# Week 3 (8 marks) dplyr, ggplot
+All the following should be done with Rmarkdown.
+
+```
+# Setup
+download.file("https://ndownloader.figshare.com/files/2292169",
+              "data/portal_data_joined.csv")
+surveys <- read.csv('data/portal_data_joined.csv')
+```
+
+1. Using two separate statements, select the columns `plot_id`, `species_id`, and `weight` from the `surveys` dataframe, and filter it to only include data from the year 1995. Do the same thing using pipes (%>%).
+2. Create a new data frame from the surveys data that meets the following criteria: Contains only the species_id column and a new column called hindfoot_half containing values that are half the hindfoot_length values. In this  hindfoot_half column, there are no NAs and all values are less than 30.
+3. Basic data aggregation.
+    a. How many individuals were caught in each plot_type surveyed?
+    b. Use group_by() and summarize() to find the mean, min, and max hindfoot length for each species (using species_id).
+    c. What was the heaviest animal measured in each year? Return the columns  year, genus, species_id, and weight.
+    d. Count the number of individuals of each sex.
+
+4. Reshape data.
```

Comment:
This question would be more useful in a later assignment. This requires some pretty challenging ways of thinking about data that the students will likely not have just yet, especially for the amount of grades this is worth. (maybe they will have the understanding.. we could always put it as a "bonus grade" or something)